test/sql/catalog/test_if_not_exists.test

```diff
@@ -14,6 +14,16 @@ CREATE TABLE IF NOT EXISTS integers2 AS SELECT 42
 statement ok
 CREATE TABLE IF NOT EXISTS integers2 AS SELECT 42
 
+statement error
+CREATE VIEW IF NOT EXISTS integers2 AS SELECT 42
+----
+Existing object integers2 is of type Table, trying to replace with type View
+
+statement error
+DROP VIEW IF EXISTS integers
+----
+Existing object integers is of type Table, trying to replace with type View
```

Comment:
"trying to replace" feels out of place in a DROP statement

Reply:
Agreed. This was the existing message in `DuckSchemaEntry::DropEntry` (not changed by this PR), I just added a test for it (to show that what we do now for create matches with what was already happening on drop). Happy to change the message in `DuckSchemaEntry::DropEntry` too if you're ok with this being in the scope of this PR. What should the message look like in your opinion?

Reply:
Thanks! Yea the changes make sense to me, good catch

Probably something like:
`Existing object "integers" is of type Table, unable to drop View` for the `DROP`
`Existing object "integers" is of type Table, unable to create View` for the `CREATE`

I hope we have enough context to distinguish between the `create` and the `drop`

Reply:
Or actually, we probably want to look at Postgres for the error messages

Reply:
Based on your feedback above I pushed [8e8a019](https://github.com/duckdb/duckdb/pull/14143/commits/8e8a01946f58c730747854bac55f9355887ebbe4).

An interesting realization from your Postgres comparison is that `create table if not exists` does not throw an exception if the existing object is not a table. Instead it ignores the comment. I've taken that same approach now.

For `DROP ... IF EXISTS` I've amended the error message to be slightly more correct. Doing a full alignment with Postgres error messages is out of scope for this PR though (I gave it a start and found plenty of inconsistencies as these errors can be thrown from multiple places).